pkg/virt-controller/watch/vmi/vmi_test.go

```diff
@@ -3557,7 +3559,58 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				"Not all of the VMI's DVs are ready",
 			),
 		)
+	})
+
+	Context("When a migration exists", func() {
+		It("should delay pod creation if the migration is running", func() {
+			vmi := newPendingVirtualMachine("testvmi")
+
+			addVirtualMachine(vmi)
+
+			migration := &virtv1.VirtualMachineInstanceMigration{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: vmi.Name + "-",
+					Namespace:    vmi.Namespace,
+				},
+				Spec: virtv1.VirtualMachineInstanceMigrationSpec{
+					VMIName: vmi.Name,
+				},
+				Status: virtv1.VirtualMachineInstanceMigrationStatus{
+					Phase: virtv1.MigrationRunning,
+				},
+			}
+			Expect(controller.migrationIndexer.Add(migration)).To(Succeed())
+
+			sanityExecute()
+			// Not expecting pod creation event, test will fail if one gets created
+			pods, err := kubeClient.CoreV1().Pods(vmi.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", virtv1.CreatedByLabel, string(vmi.UID))})
```

Comment:
nit:  You can just use plain list.

Reply:
This is a copy-paste of what `expectMatchingPodCreation()` does, made sense to me